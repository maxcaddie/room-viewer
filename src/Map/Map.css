.navigationArrows {
  color: red;
  height: 70vh;
  width: 10vw;
  font-size: 20vh;
  background-color: transparent;
}

.floorMap {
  width: 70vw;
  min-width: 600px;
  height: 80vh;
}
