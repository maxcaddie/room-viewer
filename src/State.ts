//externals

//@ts-ignore
import { createStore } from "redux";

//Actions
export function changeFloor() {
  return {
    type: "CHANGEFLOOR"
  };
}

//Reducer
const floor = (state = 0, action: any) => {
  switch (action.type) {
    case "CHANGEFLOOR":
      state = state + 1;
      return state;

    default:
      return state;
  }
};

export let store = createStore(floor);

store.subscribe(() => console.log(store.getState()));

store.dispatch(changeFloor());
