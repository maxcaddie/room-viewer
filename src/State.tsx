import { createStore, compose } from "redux";
import { nextFloor } from "./redux/actions/floorActions";
import { rootReducer } from "./redux/reducers/rootReducer";

export const store = createStore(
  rootReducer,
  compose((window as any).__REDUX_DEVTOOLS_EXTENSION__ && (window as any).__REDUX_DEVTOOLS_EXTENSION__())
);

store.dispatch(nextFloor());
