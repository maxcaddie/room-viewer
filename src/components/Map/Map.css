.container {
  width: auto;
  min-width: 600px;
  height: 90vh;
  position: relative;
  margin-top: 75px;
}

.navigationArrows {
  height: 100%;
  width: 50%;
  background-color: transparent;
  border-color: transparent;
  bottom: 0px;
  outline: none !important;
}

.left {
  position: absolute;
  left: 0px;
}

.right {
  position: absolute;
  right: 0px;
}

.floorMap {
  width: 100%;
  max-width: 110vh;
  height: 95%;
  max-height: 110vw;
}

.floor {
  font-family: "Adelle, Arial";
  color: white;
  font-size: 20px;
}
