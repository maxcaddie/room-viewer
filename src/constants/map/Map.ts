export const LEFT_ARROW_KEY_CODE = 37;
export const RIGHT_ARROW_KEY_CODE = 39;
export const A_KEY_CODE = 37;
export const D_KEY_CODE = 39;

export function isGoBack(keyCode: number) {
  return keyCode === LEFT_ARROW_KEY_CODE || keyCode === A_KEY_CODE;
}

export function isGoForward(keyCode: number) {
  return keyCode === RIGHT_ARROW_KEY_CODE || keyCode === D_KEY_CODE;
}
