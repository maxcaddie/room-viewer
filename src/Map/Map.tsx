//Internals
import { store } from "../State";
import { changeFloor } from "../redux/actions/floorActions";

//Externals
import React from "react";
import { useSelector } from "react-redux";
import { numToFloor } from "../constants/floor/Floor";

const Map: React.FC = () => {
  const floorOn = useSelector(store.getState);
  return (
    <React.Fragment>
      <button onClick={() => store.dispatch(changeFloor())}>CLICK</button>
      <p className="floor">This will say what floor you are looking at {floorOn}</p>
      <img src={numToFloor(floorOn)} alt={numToFloor(floorOn)} />
    </React.Fragment>
  );
};

export default Map;
