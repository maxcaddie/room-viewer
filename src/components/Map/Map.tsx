//Internals
import { store } from "../../State";
import { nextFloor, prevFloor } from "../../redux/actions/floorActions";
import { numToFloor, numToFloorName } from "../../constants/floor/Floor";
import { isGoBack, isGoForward } from "../../constants/map/Map";

//Externals
import React, { useEffect } from "react";
import { useSelector } from "react-redux";

//Stylesheets
import "./Map.css";
import { MOBILE_SIZE } from "../../constants/MobileView";

const Map: React.FC = () => {
  useEffect(() => {
    document.onkeydown = function(e) {
      e = e || window.event;
      if (isGoBack(e.keyCode)) {
        store.dispatch(prevFloor());
      }
      if (isGoForward(e.keyCode)) {
        store.dispatch(nextFloor());
      }
    };
  });

  const floorOn = useSelector(store.getState).floor;

  return (
    <React.Fragment>
      <style>{"body { background-color: #e9ebef; }"}</style>
      {/* <p className="floor">{numToFloorName(floorOn)}</p> */}
      <div className="container">
        <img src={numToFloor(floorOn)} alt={numToFloor(floorOn)} className="floorMap" />
        <button
          className="navigationArrows left"
          onClick={() => {
            window.innerWidth > MOBILE_SIZE ? store.dispatch(prevFloor()) : store.dispatch(nextFloor());
          }}
        />
        <button className="navigationArrows right" onClick={() => store.dispatch(nextFloor())} />
      </div>
    </React.Fragment>
  );
};

export default Map;
