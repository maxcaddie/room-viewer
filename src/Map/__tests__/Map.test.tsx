//Internals
import Map from "../Map";

//Externals
import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "../../State";

it("renders the same Map as before", () => {
  var mapComponent = renderer
    .create(
      <Provider store={store}>
        <Map />
      </Provider>
    )
    .toJSON();
  expect(mapComponent).toMatchSnapshot();
});

it("has floor explination text with floor tag", () => {
  var mapComponent = renderer.create(
    <Provider store={store}>
      <Map />
    </Provider>
  );
  mapComponent.root.findByProps({ className: "floor" });
});

describe("images load correctly", () => {
  it("has an image", () => {
    var mapComponent = renderer.create(
      <Provider store={store}>
        <Map />
      </Provider>
    );
    mapComponent.root.findByType("img");
  });
});
