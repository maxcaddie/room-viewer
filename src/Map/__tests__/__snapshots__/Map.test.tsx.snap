// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the same Map as before 1`] = `
Array [
  <button
    onClick={[Function]}
  >
     FLOORT 
  </button>,
  <button
    onClick={[Function]}
  >
    CLICK
  </button>,
  <p
    className="floor"
  >
    This will say what floor you are looking at 
    1
  </p>,
  <img
    alt="A floor"
    src="./floors/MB2.jpg"
  />,
]
`;
